src/lua/httpc.lua

```diff
@@ -317,13 +380,26 @@ curl_mt = {
             if not method or not url then
                 error('request(method, url[, body, [options]])')
             end
+
+            opts = opts or {}
+            opts.headers = opts.headers or {}
+            method = method:upper()
+            if method == 'PATCH' or
+               method == 'POST' or
+               method == 'PUT' then
+                local content_type = opts.headers['content-type']
```

Comment:
1. Header names are case insensetive (it is from my memory, but I'm sure there is a formal wording).
2. Content-type values (type/subtype) are also case insensetive (according to RFC 1521, see section 4, comments in the BNF).

Reply:
If I read the code correctly, it doesn't handle the second point, e.g. `Application/JSON`. Reopened the discussion to re-check.